src/services/mpris/mod.rs

```diff
@@ -0,0 +1,340 @@
+                MprisPlayerProxy::new(conn, n.clone())
+                    .await
+                    .inspect_err(|e| error!("Failed to connect MPRIS player proxy: {e}"))
+            }))
+            .await
+            .iter()
+            .filter_map(|r| r.clone().ok())
+            .collect();
+        for s in services.iter() {
+            combined.push(
+                s.receive_metadata_changed()
+                    .await
+                    .map(|_| Event::Metadata)
+                    .boxed(),
+            );
+        }
+        for s in services.iter() {
+            combined.push(
+                s.receive_volume_changed()
+                    .await
+                    .map(|_| Event::Volume)
+                    .boxed(),
+            );
+        }
+        Ok(combined)
+    }
+
+    async fn start_listening(state: State, output: &mut Sender<ServiceEvent<Self>>) -> State {
+        match state {
+            State::Init => match zbus::Connection::session().await {
+                Ok(conn) => {
+                    let data = Self::initialize_data(&conn).await;
+                    match data {
+                        Ok((names, data)) => {
+                            info!("MPRIS player service initialized");
+
+                            let _ = output
+                                .send(ServiceEvent::Init(MprisPlayerService {
+                                    data,
+                                    conn: conn.clone(),
+                                }))
+                                .await;
+
+                            State::Active(conn, names)
+                        }
+                        Err(err) => {
+                            error!("Failed to initialize MPRIS player service: {}", err);
+
+                            State::Error
+                        }
+                    }
+                }
+                Err(err) => {
+                    error!("Failed to connect to system bus for MPRIS player: {}", err);
+                    State::Error
+                }
+            },
+            State::Active(conn, names) => match Self::events(&conn).await {
+                Ok(mut events) => {
+                    let mut names = names;
+                    while let Some(event) = events.next().await {
+                        match event {
+                            Event::NameOwner => match Self::initialize_data(&conn).await {
+                                Ok(data) => {
+                                    info!("MPRIS player service new data");
+                                    names = data.0;
+                                    let _ = output.send(ServiceEvent::Update(data.1)).await;
+                                }
+                                Err(err) => {
+                                    error!("Failed to fetch MPRIS player data: {}", err);
+                                }
+                            },
+                            _ => {
```

Comment:
```suggestion
                            Event::Volume | Event::Metadata => {
```
I prefer to be more explicit :sweat_smile:

Reply:
Fair, I do too 🙂